### ISSUE TITLE
Minecraft versions : branches or folder

### ISSUE THREAD

**Issue by rom1504** [PrismarineJS/minecraft-data#64]
Currently mc version are stored as different branches.

Problem with that is a lot of commits need to be cherry-picked from 1.8 to 1.9 because they change things common between versions : changes in the README, sometimes changes in the extractors, new data files.

I think storing version in enums/1.8 and enums/1.9 might be saner : only the data should really be different.
* the schemas should be the same : see  #49 
* the readme should be the same
* the extractor and tools are a bit more complicated, but basically it would be better to have everything that is common in common files and to have a few functions/files specific for different mc version / revision of the wiki / java decompiled code format

I think to be really sure it can work, the best way would be to try to use the extractors for 1.9.

So I'll do that at some point and that'll give us more info into that issue.

**Comment by roblabla** [comment]
an alternative to this would be putting back a master branch with just the extractors, the README, and have the readme link to the version branches. IDK which is better, a part of me likes the way things are organized right now, but it does incur some maintenance overhead.

**Comment by rom1504** [comment]
https://github.com/PrismarineJS/minecraft-data/issues/65#issuecomment-142259535 convinces me even more that the wiki extractor really shouldn't be in minecraft-data.
Their output format is mcData schema yes, but apart from that they are independant from minecraft-data. I think I might try putting them in a separate repo, what would be nice about that is treating these extractors as really a first class package, and doing things like automatically following the wiki changes and having that "wiki watcher" bot do PR to minecraft-data or something.

About a master branch : a possibility would be having everything but the data (the schema, the auto-testing, (possibly the extractors if they stay there), the readme) in master. And version branches would just contain the data (the .json files).
It would probably mean submoduling these version branches in master which has pros (no need to do this in each X-minecraft-data) and cons (need to update the submodule often to check the data fits the schema). Other ways than submoduling can be considered (in order to check the schemas for example).

The general idea here is to separate things that are independent and to do as little copy pasting (~= cherry-pick) as possible.

**Comment by roblabla** [comment]
I _really think_ the extractors should be in the minecraft-data repo. It allows anyone to easily see how the data was generated, and easily find the tools to generate it themselves. The alternative, putting links to all the tools, is OK, but we need to make sure we keep that list up to date,

**Comment by rom1504** [comment]
So, I did everything in this issue: put the extractors in different repos (listed in the readme) and put everything in master branch to avoid duplication.
I changed everything on projects that depend on minecraft-data (directly in nmd and pmd and via PR in ProtocolGen and phase_bot)
I'll let 1.8 and 1.9 branches live for a little while more, so phase_bot can merge the pr and other potential users can use master and then I'll delete them.
I'll close this issue then.
